logservice/logpuller/subscription_client.go

```diff
@@ -232,7 +232,10 @@ func NewSubscriptionClient(
 	option := dynstream.NewOption()
 	// Note: it is max batch size of the kv sent from tikv(not committed rows)
 	option.BatchCount = 1024
-	option.UseBuffer = false
+	// TODO: only set UseBuffer to false when regionEventHandler.Handle won't call any method of dynamic stream.
```

Comment:
```suggestion
	// TODO: Set `UseBuffer` to true until we refactor the `regionEventHandler.Handle` method so that it doesn't call any method of the dynamic stream. Currently, if `UseBuffer` is set to false, there will be a deadlock:
```